agent/dockerclient/dockerauth/asm.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Does it make sense to restructure this folder to move each auth provider into its own folder? like:
```
agent/dockerclient/dockerauth/docker
agent/dockerclient/dockerauth/ecr
agent/dockerclient/asm
```

Reply:
hm im not sure. i thought about that but then felt maybe 3 different ones weren't enough to justify further breakdown into sub packages. but i'm not sure though.